test/kafo_parsers/puppet_module_parser_test.rb

```diff
@@ -5,6 +5,9 @@
 module KafoParsers
   describe PuppetModuleParser do
     describe ".parse(class)" do
+      # we skip tests for this parser if puppet 4.0.0+ is installed
+      next if Gem::Specification.find_all_by_name('puppet').sort_by(&:version).last.version >= Gem::Version.new('4.0.0')
```

Comment:
I think that at this point we haven't required puppet gem, it's after first parser initialization now